src/System.ServiceProcess.ServiceController/tests/ServiceBaseTests.cs

```diff
@@ -174,6 +174,38 @@ public void TestOnContinueBeforePause()
             Assert.Equal(expected, _testService.GetServiceOutput());
         }
 
+        [ConditionalFact(nameof(IsProcessElevated))]
+        public void LogWritten()
+        {
+            using (EventLog eventLog = new EventLog("Application"))
+            {
+                ServiceBase sb = new ServiceBase() { ServiceName = "hello" };
+                Assert.False(EventLog.SourceExists(sb.ServiceName));
+                int count = eventLog.Entries.Count;
+                ServiceBase.Run(sb);
+                eventLog.Source = sb.ServiceName;
+                Assert.True(EventLog.SourceExists(sb.ServiceName));
+                Assert.True(eventLog.Entries.Count > count);
```

Comment:
Based on our experience of the event log, I have a feeling this is going to be flaky. You might want to just remove the check of count.. even though then you're not really testing writing...

Reply:
We can remove the count check when AutoLog is False as there could be more apps writing to Application Log. But I think we should keep the count check when the AutoLog is true.
 The reason If the Assert Fails then we can be sure the log has not been written